src/components/DataDownloader/DataDownloader.tsx

```diff
@@ -0,0 +1,57 @@
+import { FC, HTMLAttributes, useEffect, useRef, useState } from 'react';
```

Comment:
I have a hook I made for the export component that handles downloading files if it would be of use here

https://github.com/adsabs/nectar/blob/master/src/hooks/useDownloadFile.ts